blog/views.py

```diff
@@ -44,4 +44,35 @@ def form_valid(self, form):
         return super().form_valid(form)
     
     def get_success_url(self):  
-        return reverse("blog:blog-detail", kwargs={"pk": self.kwargs["pk"],})
\ No newline at end of file
+        return reverse("blog:blog-detail", kwargs={"pk": self.kwargs["pk"],})
+
+class CreateBlogView(LoginRequiredMixin, CreateView):
+    template_name = "blog/create_blog.html"
+    model = Blog
+    fields = ["title","content"]
+    
+    def form_valid(self, form):
+        new_blog = form.save(commit=False)
+        new_blog.author = self.request.user
+        return super().form_valid(form)
+    
+    def get_success_url(self): 
+        return reverse("author:author-profile",kwargs={"pk":self.request.user.id})
+
+
+class UpdateBlogView(LoginRequiredMixin, UpdateView):
+    template_name = "blog/update_blog.html"
+    model = Blog
+    fields = ["title","content"]
+
+    def get_success_url(self):
+        return reverse("author:author-profile",kwargs={"pk":self.request.user.id})
+
+
+class DeleteBlogView(LoginRequiredMixin, DeleteView):
+    template_name = "blog/delete_blog.html"
+    model = Blog
+    context_object_name = "post"
+    
+    def get_success_url(self):
+        return reverse("author:author-profile",kwargs={"pk":self.request.user.id})
```

Comment:
Requested user can only update and delete their created blogs only. But in this code, any user can update and delete others user's blog.